packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -60,6 +65,69 @@ export const defaultServices = {
 
   // Validation hooks for overriding
   async validateCustomSignUp(formData, touchData): Promise<ValidatorResult> {},
+  async validateFormPassword<Validator>(
+    formData,
+    touchData,
+    passwordSettings: PasswordSettings
+  ): Promise<ValidatorResult> {
+    const { password } = formData;
+
+    const { password: touched_password } = touchData;
+
+    /**
+     * If the password is not touched,
+     * or if the password settings are not set, we don't need to validate it.
+     */
+    if (!touched_password || !passwordSettings) return null;
+
+    const password_complexity = [];
+
+    const policyMinLength = +passwordSettings?.passwordPolicyMinLength;
+    if (password.length < policyMinLength) {
+      password_complexity.push(
+        `Password must contain at least ${policyMinLength} characters`
+      );
+    }
+
+    const passwordPolicyCharacters = passwordSettings?.passwordPolicyCharacters;
+
+    passwordPolicyCharacters?.forEach((errorCheck) => {
+      switch (errorCheck) {
+        case 'REQUIRES_LOWERCASE':
+          if (!/[a-z]/.test(password))
```

Comment:
Just using regex here for all these checks. Maybe there is a better way... not sure.